ddtrace/internal/datadog/profiling/ddup/CMakeLists.txt

```diff
@@ -49,24 +49,13 @@ add_custom_command(
 # Specify the target C-extension that we want to build
 add_library(${EXTENSION_NAME} SHARED ${DDUP_CPP_SRC})
 
-# We can't add common Profiling configuration because cython generates messy code, so we just setup some basic flags and
-# features
-target_link_options(${EXTENSION_NAME} PRIVATE "$<$<CONFIG:Release>:-Os>" -ffunction-sections)
-
-if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
-    # macOS-specific options
-    target_compile_options(${EXTENSION_NAME} PRIVATE "$<$<CONFIG:Debug>:-Og;-g>")
-    target_link_options(${EXTENSION_NAME} PRIVATE "$<$<CONFIG:Release>:-Wl,-dead_strip>")
-else()
-    # Linux/ELF-based options
-    target_compile_options(${EXTENSION_NAME} PRIVATE "$<$<CONFIG:Debug>:-Og;-ggdb3>" -fno-semantic-interposition)
-    target_link_options(${EXTENSION_NAME} PRIVATE "$<$<CONFIG:Release>:-s>" -Wl,--as-needed -Wl,-Bsymbolic-functions
-        -Wl,--gc-sections)
-endif()
-
-set_property(TARGET ${EXTENSION_NAME} PROPERTY INTERPROCEDURAL_OPTIMIZATION TRUE)
-
-target_compile_features(${EXTENSION_NAME} PUBLIC cxx_std_17)
+add_ddup_config(${EXTENSION_NAME})
+# Cython generates code that produces errors for the following, so relax compile
+# options
+target_compile_options(
+    ${EXTENSION_NAME}
+    PRIVATE -Wno-old-style-cast -Wno-shadow
```

Comment:
Based on this build error, do we also want `-Wno-address` here?

https://github.com/DataDog/dd-trace-py/actions/runs/11936849079/job/33278388183?pr=11457#step:7:6948
```
[ 93%] Building CXX object CMakeFiles/_ddup.cpython-39-x86_64-linux-gnu.so.dir/_ddup.cpp.o
      /tmp/pip-req-build-t9uc854h/build/cmake.linux-x86_64-cpython-39/ddtrace.internal.datadog.profiling.ddup._ddup/_ddup.cpp: In function 'PyObject* __Pyx_Import(PyObject*, PyObject*, int)':
      /tmp/pip-req-build-t9uc854h/build/cmake.linux-x86_64-cpython-39/ddtrace.internal.datadog.profiling.ddup._ddup/_ddup.cpp:14745:48: error: comparing the result of pointer addition '(((const char*)"ddtrace.internal.datadog.profiling.ddup._ddup") + 7)' and NULL [-Werror=address]
      14745 |             if (strchr(__Pyx_MODULE_NAME, '.') != NULL) {
            |                                                ^
      cc1plus: all warnings being treated as errors
```

(looks like the `strchr` call is being optimized at compile time since the input string is known)

Reply:
Good catch. Updated.